project-icp/bee-pollinator-app
# Apply hive density and demand values to model

### Overview
Crop density (recommend hives per acre) and demand (dependence on pollination) are specific per crop, but have not previously been provided to the model.  This PR updates and supplies the actual crop values and not the temporary placeholders used to this point.  I chose to keep the text description for demand in the base CSV so that adjusting the numerical values later would be easier.

Connects #120 
Connects #144 

Conceptually, this means now that when the user applies a manged hive per acre value to a scenario, it does not uniformly increase the yield to all crops.  In the case of a scenario with two crops Orchard (1.53) and Almond (6), raising the hive value will bring Orchard yield to nearly 100 and have a more modest effect on Almond (unless you've supplied wild pollinator habitat to increase yield).

#### Orchard and Almond yield at 1.5 hives per acre
![screenshot from 2017-02-09 09 57 59](https://cloud.githubusercontent.com/assets/1014341/22790289/b307fc72-eeb3-11e6-81b6-bbeac32117c9.png)

Demand was previously an arbitrary number, but it was still dynamically allocated across crops - now that we are using the real number, the yield for a crop will still be variable, but the variable is now the actual demand (not the crop id).  Demand can be thought of as how sensitive yield is to the presence of pollinator species, and each crop has a different sensitivity.

### Testing
* Update the installed model 
```
vagrant ssh worker -c "cd /opt/app/pollinator && sudo python setup.py install"
```
* Tests and lint pass
```
./scripts/test.sh
```
* Verify that the recommended hive amounts correlate to the reported yield for a crop in the app.
See "density" column: https://github.com/project-icp/bee-pollinator-app/blob/develop/src/icp/pollinator/src/pollinator/data/cdl_data_grouped.csv
